sys/trace/Kconfig

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config MODULE_TRACE
+    bool "Execution tracing"
+    depends on TEST_KCONFIG
+    select MODULE_XTIMER
+
+if MODULE_TRACE
+
+config TRACE_BUFSIZE
+    int "Trace buffer size"
+    default 512
+
+endif # MODULE_TRACE
```

Comment:
For this we can take the approach of #16037 and splitting the configuration option, so one can use it both while testing dependency modelling and in normal Kconfig usage.